Mathlib/Data/Nat/Nth.lean

```diff
@@ -243,22 +242,23 @@ theorem isLeast_nth_of_infinite (hf : (setOf p).Infinite) (n : ℕ) :
   isLeast_nth fun h => absurd h hf
 #align nat.is_least_nth_of_infinite Nat.isLeast_nth_of_infinite
 
-/-- An alternative recursive definition of `nat.nth`: `nat.nth s n` is the infimum of `x ∈ s` such
-that `nat.nth s k < x` for all `k < n`, if this set is nonempty. We do not assume that the set is
-nonempty because we use the same "garbage value" `0` both for `Inf` on `ℕ` and for `nat.nth s n` for
+/-- An alternative recursive definition of `Nat.nth`: `Nat.nth s n` is the infimum of `x ∈ s` such
+that `Nat.nth s k < x` for all `k < n`, if this set is nonempty. We do not assume that the set is
+nonempty because we use the same "garbage value" `0` both for `Inf` on `ℕ` and for `Nat.nth s n` for
 `n ≥ card s`. -/
 theorem nth_eq_sInf (p : ℕ → Prop) (n : ℕ) : nth p n = sInf {x | p x ∧ ∀ k < n, nth p k < x} := by
-  by_cases hn : ∀ hf : (setOf p).Finite, n < hf.to_finset.card
-  · exact (is_least_nth hn).csInf_eq.symm
-  · push_neg at hn 
+  by_cases hn : ∀ hf : (setOf p).Finite, n < hf.toFinset.card
+  · exact (isLeast_nth hn).csInf_eq.symm
+  · -- porting note: TODO: was push_neg at hn
+    simp only [not_forall, not_lt] at hn
```

Comment:
`push_neg` should work better now

Reply:
Because `push_neg at hn` makes a loose bounded variable.
I'm going to fix this bug.

Reply:
#5257